assets/js/components/settings/settings-modules.js

```diff
@@ -82,45 +81,43 @@ class SettingsModules extends Component {
 	 *
 	 * @since 1.0.0
 	 *
-	 * @param {string} module The module slug.
-	 * @param {string} action The action being performed, one of 'edit', 'cancel' or 'confirm'.
+	 * @param {string}   module        The module slug.
+	 * @param {string}   action        The action being performed, one of 'edit', 'cancel' or 'confirm'.
+	 * @param {Function} submitChanges The action dispatcher to submit the changes.
 	 */
-	handleButtonAction( module, action ) {
+	async handleButtonAction( module, action, submitChanges ) {
 		if ( 'confirm' === action ) {
-			const modulePromise = applyFilters( 'googlekit.SettingsConfirmed', false, module );
-
 			this.setState( { isSaving: module } );
-			if ( ! modulePromise ) {
-				// Clears session and local storage on successful setting.
-				clearWebStorage();
-
-				return;
-			}
-			modulePromise.then( () => {
-				// Clears session and local storage on every successful setting.
-				clearWebStorage();
-
-				this.setState( {
-					isSaving: false,
-					error: false,
-				} );
+			const { error } = await submitChanges( module );
 
-				this.props.setModuleState( 'view' );
-			} ).catch( ( err ) => {
-				let error;
-				if ( isPermissionScopeError( err ) ) {
-					error = false;
+			if ( error ) {
+				if ( isPermissionScopeError( error ) ) {
+					this.setState( {
+						error: false,
+					} );
 				} else {
-					error = {
-						errorCode: err.code,
-						errorMsg: err.message,
-					};
+					this.setState( {
+						error: {
+							errorCode: error.code,
+							errorMsg: error.message,
+						},
+					} );
 				}
 				this.setState( {
 					isSaving: false,
-					error,
 				} );
```

Comment:
Let's not make more `setState` calls here than needed, we can change this code to always make a single one, which is also what the previous implementation did. Since `isSaving` is always set to `false`, let's put that into both `setState` calls in the clause above and remove the call from here instead.